front/src/reducers/osrdconf/operationalStudiesConf/powerRestrictionReducer.ts

```diff
@@ -163,42 +163,31 @@ const powerRestrictionReducer = {
     action: PayloadAction<{
       firstRestriction?: PowerRestriction;
       secondRestriction: PowerRestriction;
+      endPosition: number;
       newFromPathStep: PathStep;
     }>
   ) {
-    const { firstRestriction, secondRestriction, newFromPathStep } = action.payload;
+    const { firstRestriction, secondRestriction, endPosition, newFromPathStep } = action.payload;
 
     // pathSteps should not be undefined or have null values
-    if (state.pathSteps && !state.pathSteps.some((pathStep) => !pathStep)) {
+    if (!state.pathSteps.some((pathStep) => !pathStep)) {
       let newPathSteps = [...state.pathSteps].map((step) => step!);
-      let newPowerRestrictionRanges = state.powerRestriction.filter(
-        (restriction) =>
-          !isEqual(restriction, firstRestriction) || !isEqual(restriction, secondRestriction)
-      );
 
       // find the covered ranges
-      const pathStepEnd = newPathSteps.find((pathStep) => pathStep.id === secondRestriction.to);
-      const coveredRanges = pathStepEnd
-        ? newPowerRestrictionRanges.filter((restriction) =>
-            isRangeCovered(
-              newPathSteps,
-              restriction,
-              newFromPathStep.positionOnPath,
-              pathStepEnd.positionOnPath
-            )
-          )
-        : [];
+      const coveredRanges = [...state.powerRestriction].filter((restriction) =>
```

Comment:
`filter` already returns a new array (same for `resizeSegmentEndInput`)
```suggestion
      const coveredRanges = state.powerRestriction.filter((restriction) =>
```

Reply:
Done